cub/cub/device/dispatch/dispatch_histogram.cuh

```diff
@@ -773,7 +761,9 @@ public:
     _CCCL_HOST_DEVICE _CCCL_FORCEINLINE int
     ComputeBin(T sample, T min_level, ScaleT scale, ::cuda::std::false_type /* is_fp */)
     {
-      return static_cast<int>(((sample - min_level) * scale.fraction.bins) / scale.fraction.range);
+      return static_cast<int>(::cuda::std::divides<T>{}(
+        ::cuda::std::multiplies<T>{}(::cuda::std::minus<T>{}(sample, min_level), scale.fraction.bins),
+        scale.fraction.range));
```

Comment:
critical: this yields different result in presence of integer promotion. I don't think we can make this change:
```cpp
  char sample = 100;
  char min_level = 0;
  char max_level = 120;
  CommonT bins = 120;
  CommonT range = max_level - min_level;
  std::cout << OldComputeBin(sample, min_level, bins, range) << std::endl; // 100
  std::cout << NewComputeBin(sample, min_level, bins, range) << std::endl; // 0
```